codex/marketplace.md

```diff
@@ -0,0 +1,209 @@
+The goal is to create a storage marketplace that promotes durability.
+
+## Motivation
+The Codex network aims to create a peer-to-peer storage engine with strong data durability, 
+data persistence guarantees and node storage incentives.
+To reach this goal, a data availability and retrieval mechanism is needed.
+Support for light clients, like mobile devices should also be embraced.
+The protocol should remove complexity to allow for simple implementation and 
+simplify incentive mechanisms.
+
+## Semantics 
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+| Terminology  | Description |
+| --------------- | --------- |
+| storage providers | A Codex node that provides storage services to the marketplace. |
+| validator nodes | A Codex node that collects, validates and submits missing storage proofs for a reward. |
+| client nodes | The most common Codex node that interacts with other nodes to store, locate and retrieve data. Also considered to be an ephemeral node (light client). |
+| slots | Created by client nodes when a new dataset is requested to be stored. Discussed further in the [slots section](#slots).  |
+
+### Storage Request
+
+Client nodes can create storage requests on the Codex network via the Codex marketplace.
+The marketplace is a a smart contract on a EVM compatiable blockchain.
+The smart contract SHOULD handle storage request, storage slot state, 
+storage provider rewards, storage provider collateral, and storage proof state. 
+
+To create a request to store a dataset on the Codex network,
+client nodes MUST split the dataset into data chunks, $(c_1, c_2, c_3, \ldots, c_{20})$.
+Using a erasure coding technique, 
+the data chunks are encoded and placed into separate slots.
+The RECCOMENDED erasure coding technique is a [Reed-Soloman algorithm](https://hackmd.io/FB58eZQoTNm-dnhu0Y1XnA).
+
+The requester, client node, SHOULD submit a transaction with the desired request parameters.
+The requester MUST provide a `duration` value for the storage request along with the appropriate `reward`,
+payment for request. 
+Once a request is created via the transaction, 
+all slots MUST be filled by storage providers before the request is officially started.
+If the request does not attract enough storage providers after a pre-defined network timeout,
+which is defined by the Codex node as `expiry`,
+the request MUST be canceled.
+If canceled, `collateral` SHOULD be returned to any storage providers and 
+`reward` returned to the requester.
+The requester MAY create a new request with different values to restart the process.
+
+The storage request SHOULD initiate a new storage contract with the following parameters:
```

Comment:
All these parameters are required so most probably it should be `MUST`?

Reply:
`MAY` in this context refers to the requester's choice of creating a new transaction. `SHOULD` in this context refers to the requester providing the values for the request. In most cases a `MUST` is used for described the behavior of the application itself not the user because a user may not provide all the values.